pkg/resourceinterpreter/configurableinterpreter/configurable.go

```diff
@@ -30,128 +31,124 @@ func NewConfigurableInterpreter(informer genericmanager.SingleClusterInformerMan
+	luaScript, enabled := c.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretStatus)
+	if !enabled {
+		return
 	}
-
-	luaScript := customAccessor.GetReplicaRevisionLuaScript()
-	klog.Infof("lua script %s", luaScript)
-
-	return nil, nil
+	vm := luavm.VM{UseOpenLibs: false}
+	revised, err = vm.ReviseReplica(object, replica, luaScript)
+	return
 }
 
 // Retain returns the objects that based on the "desired" object but with values retained from the "observed" object.
-func (c *ConfigurableInterpreter) Retain(desired *unstructured.Unstructured, observed *unstructured.Unstructured) (retained *unstructured.Unstructured, err error) {
-	klog.Infof("ConfigurableInterpreter Execute Retain")
-	customAccessor := c.configManager.LuaScriptAccessors()[desired.GroupVersionKind()]
-	if len(customAccessor.GetRetentionLuaScript()) == 0 {
-		return nil, fmt.Errorf("customized interpreter operation Retain for %q not found", desired.GroupVersionKind())
+func (c *ConfigurableInterpreter) Retain(desired *unstructured.Unstructured, observed *unstructured.Unstructured) (retained *unstructured.Unstructured, enabled bool, err error) {
+	luaScript, enabled := c.getInterpreter(desired.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretStatus)
+	if !enabled {
+		return
 	}
-
-	luaScript := customAccessor.GetRetentionLuaScript()
-	klog.Infof("lua script %s", luaScript)
-
-	return nil, err
+	vm := luavm.VM{UseOpenLibs: false}
+	retained, err = vm.Retain(desired, observed, luaScript)
+	return
 }
 
 // AggregateStatus returns the objects that based on the 'object' but with status aggregated.
-func (c *ConfigurableInterpreter) AggregateStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (*unstructured.Unstructured, error) {
-	klog.Infof("ConfigurableInterpreter Execute AggregateStatus")
-	customAccessor := c.configManager.LuaScriptAccessors()[object.GroupVersionKind()]
-	if len(customAccessor.GetStatusAggregationLuaScript()) == 0 {
-		return nil, fmt.Errorf("customized interpreter AggregateStatus Retain for %q not found", object.GroupVersionKind())
+func (c *ConfigurableInterpreter) AggregateStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (status *unstructured.Unstructured, enabled bool, err error) {
+	luaScript, enabled := c.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretStatus)
+	if !enabled {
+		return
+	}
+	vm := luavm.VM{UseOpenLibs: false}
+
+	var aggregateItem []map[string]interface{}
+	for _, item := range aggregatedStatusItems {
+		if item.Status == nil {
+			continue
+		}
+		temp := make(map[string]interface{})
+		if err = json.Unmarshal(item.Status.Raw, &temp); err != nil {
+			return
+		}
+		aggregateItem = append(aggregateItem, temp)
```

Comment:
This processing logic looks extra (unlike other operations), and we may need to pass in the entire object list, as the user may need other info(such as clusterName and health) for customization.

Just as default interpreter behavior:

https://github.com/karmada-io/karmada/blob/bc7b298c4073cd6827255741aab3ce6ae708e54a/pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go#L102-L122